### PR TITLE
feat: Add --locked flag to build and run

### PR DESCRIPTION
**Pull request by Jesse-Bakker**
Depends on #1945

### CODE REVIEW

File: dozer-cli/src/simple/build/contract/mod.rs [hunk 1]
```diff
@@ -239,4 +240,17 @@ where
     serde_json::from_reader(file).map_err(BuildError::FailedToLoadExistingContract)
 }
 
+impl PartialEq<Contract> for Contract {
+    fn eq(&self, other: &Contract) -> bool {
+        self.version == other.version
```
Comment (chubei) [inline dozer-cli/src/simple/build/contract/mod.rs:245]: This worries me a little that when we add new fields to `Contract` we'll forget to update `PartialEq` implementation.

Maybe we can do
```rust
let Contract {
  version: self_version,
  pipeline: self_pipeline,
  endpoint: self_endpoints,
} = self;
let Contract {
  version: other_version,
  pipeline: other_pipeline,
  endpoint: other_endpoints,
} = other;
```

And then compare them.

**Review by chubei — APPROVED** [review]
`is_isomorphic_matching` is handy!

File: dozer-cli/src/simple/build/contract/mod.rs (hunk 1 above)
Comment (Jesse-Bakker) [reply dozer-cli/src/simple/build/contract/mod.rs:245]: Or maybe wrap pipeline in a newtype and impl partialeq on that and derive it on Contract?

Comment (chubei) [reply dozer-cli/src/simple/build/contract/mod.rs:245]: Even better

Comment (chubei) [reply dozer-cli/src/simple/build/contract/mod.rs:245]: The new type is called `PipelineContract` by the way. It's already a alias type.

**Pull request merged**